### ISSUE TITLE
Parsing import allows "as" before "if"

### ISSUE THREAD

**Issue by munificent** [dart-lang/sdk#56641]
I don't get any errors in:

```dart
import 'foo.dart' as foo if (dart.library.html) 'other_foo.dart';

main() {
  print(foo.bar);
}
```

This runs if I run it with `dart temp.dart` and analyzes without error using `dart analyze`. The grammar for imports is:

> ⟨importSpecification⟩ ::=
>   **import** ⟨configurableUri⟩ (**deferred**? **as** ⟨typeIdentifier⟩)? ⟨combinator⟩*
>  ‘;’

That means that the `as` clause, if present, must be *after* any `if` configurations.

I ran into this issue because `dart format` will always output the `as` clause after the `if` configurations, which then trips up its sanity check because it made a non-whitespace change by reordering those clauses.

I did find at least one library in the wild (sentry-8.3.0/lib/src/environment/environment_variables.dart) that has the clauses in the wrong order, so fixing this is both nominally and actually a breaking change (though likely a very minor one given the rarity of conditional imports).

**Comment by munificent** [comment]
> I ran into this issue because `dart format` will always output the `as` clause after the `if` configurations, which then trips up its sanity check because it made a non-whitespace change by reordering those clauses.

I have a PR out for the formatter (https://github.com/dart-lang/dart_style/pull/1555) which will simply preserve whatever order the input code had the clauses in, so the formatter isn't blocked on having this fix. It will gracefully handle imports whether they are spec compliant or not.
